### PR TITLE
Generalize Next.js example into next() macro and add working Next.js devserver

### PR DESCRIPTION
**Pull request by gregmagolan**
Also removes swc usage to not overcomplicate this example and distract from the material Next.js portions

### CODE REVIEW

File: next.js/bazel/next_entry.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
+const path = require('path');
```
Comment (alexeagle) [inline next.js/bazel/next_entry.js:1]: high-level comment about why we have a custom entry point?

**Review by alexeagle — COMMENTED** [review]
shouldn't there be more files deleted under /pages ?

File: next.js/BUILD.bazel [hunk 2]
```diff
@@ -1,121 +1,10 @@
-load("@aspect_bazel_lib//lib:copy_to_bin.bzl", "copy_to_bin")
-load("@npm//:next/package_json.bzl", next_bin = "bin")
-load("@npm//:defs.bzl", "npm_link_all_packages")
-load("@aspect_rules_js//js:defs.bzl", "js_binary", "js_run_binary", "js_test")
 load("@aspect_rules_ts//ts:defs.bzl", "ts_config")
+load("@npm//:defs.bzl", "npm_link_all_packages")
 
 npm_link_all_packages(name = "node_modules")
```
Comment (alexeagle) [inline next.js/BUILD.bazel:4]: is this used?

**Review by alexeagle — APPROVED** [review]

File: next.js/BUILD.bazel (hunk 2 above)
Comment (gregmagolan) [reply next.js/BUILD.bazel:4]: yes. there is a pnpm workspace so the virtual store is linked created at root as well as some of devDependencies

File: next.js/app/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    ],
+    data = [
+        "next.config.js",
+        ":node_modules/next",
+        ":node_modules/react",
+        ":node_modules/react-dom",
+        ":package.json",
+        "//:node_modules/typescript",
+    ],
+    next_bin = "./node_modules/.bin/next",
```
Comment (jbedard) [inline next.js/app/BUILD.bazel:32]: I guess a given BUILD will have it's own nextjs so we can't put this attribute into the macro?

File: next.js/bazel/next_entry.js [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+
+const path = require('path');
 const { spawnSync } = require('child_process');
-// This path is currently a hack an specific to this projects layout with
-// ../../../../../ being the path from the the next_bin binary runfiles root
-// to the execroot.
-// TODO: Generalize this path in the future.
 const entry = require.resolve(
-  `../../../../../${process.env.BAZEL_BINDIR}/node_modules/next/dist/bin/next`
+  path.join(
```
Comment (jbedard) [inline next.js/bazel/next_entry.js:7]: Could this path be replaced with `"$(location %s)" % next_bin` or something like that in the BUILD?

Comment (gregmagolan) [reply next.js/bazel/next_entry.js:7]: `$(location)` would require also passing the target to `data` which I'm avoiding here. This is temporary anyway as I'm going to add a `command` attribute to `js_run_binary`  which will make this custom entry point boil away

**Comment by gregmagolan** [comment]
> shouldn't there be more files deleted under /pages ?

🤔 `pages` was moved to `app/pages`

**Pull request merged**